### DISCUSSION TITLE
Linking GPS to video frames

### DISCUSSION THREAD

**Discussion by jakobsnl** [gopro/labs#426] [General]
Does someone have any experience with linking up GPS time stamps with video frames? And more precisely, is there a way to obtain the distance beetween the first GPS timestamp and the first image frame in milliseconds accuracy? Many thanks in advance! Jakob

**Comment by dnewman-gpsw** [comment]
GPS and video metadata are timestamped with microsecond precision with the GPMF data.  Use tools from https://github.com/gopro/gpmf-parser

**Reply by jakobsnl** [reply]
@dnewman-gpsw As far as i can analyse the meta data is not saved  in milliseconds accuracy? 
If that is wrong, how can a millisecond time stamp of the create time of a video be obtained?

**Reply by dnewman-gpsw** [reply]
The STMP metadata field is the microsecond precision for each payload.  STMP is the timestamp for the first sample in each payload, so you can interpolate the STMP across all the samples.  GPS9 contains only millisecond precision for the value of time, yet it shifts its output to be in sync with the extract clock, so that the time is mostly 00 in the end.  It is little surprising, but that is what the GPS part seems to do.
